### DISCUSSION TITLE
How can I use the url() method with a bulk action?

### DISCUSSION THREAD

**Discussion by crosa7** [filamentphp/filament#13260] [Help]
### Package

Panel builder

### Package Version

v3.2.14

### How can we help you?

Hey, hey!

I am trying to call a url with a `BulkAction` but I am not able to do it, as it seems that the `$records` always come empty with the `url()` method, but with the `action()` they come filled with the selected records correctly.
Is this not possible or I am doing something wrong? Thanks in advance

This is what I tried:

```
$bulkAction
        ->url(function (?Collection $records) {
            $uuids = array_map(fn (array $orderData) => $orderData['uuid'], $records->toArray());

            return route('pdf', ['uuids' =>  $uuids]);
        });
```

**Comment by christmex** [comment]
Hi @crosa7, sorry if I misunderstood you, but maybe you try this one,

if you use bulk action, I think you can pass the param using this one ``` $livewire ``` then to get all of the records use this ``` $livewire->getSelectedTableRecords() ``` 


Thanks, let me know if you find any issue

**Reply by crosa7** [reply]
Thank you for the help @christmex, but unfortunately it still gives an empty collection. I am thinking that this might really not be possible, as the url will render before any records on the table are selected, so when I click on it I have no `uuids` as parameters because the url was rendered when the table loads and no records are selected yet of course

**Reply by olivM** [reply]
i have the same need, did you find a way to achieve this ?

**Reply by crosa7** [reply]
In the end I just made an action with a redirect inside:

`$bulkAction
            ->action(function (Collection $records) {
                redirect()->route('pdf');
            });`
